Update main.py (More print statments, and command errors)

blobonmyknob opened the pull request.

blobonmyknob committed:
Update main.py (Added more print statements.)

helish88 commented on main.py:
```diff
@@ -1,34 +1,42 @@
-@bot.command(aliases=['start', 'g'])
-@commands.has_permissions(manage_guild=True)
+@bot.command(aliases = ['start , g'])
+@commands.has_permissions(manage_guild = True)
```
Why does a bot need this permissions?

blobonmyknob replied:
The bot doesn't it's for the user, if the user has the permission, they can use the command otherwise it returns an error. :/

helish88 replied:
right, sry.

blobonmyknob replied:
so this is no longer an error

helish88 commented on main.py:
```diff
@@ -1,34 +1,42 @@
 # Giveaway Command
-@bot.command(aliases=['start', 'g'])
-@commands.has_permissions(manage_guild=True)
+@bot.command(aliases = ['start , g'])
```
It's wrong

blobonmyknob replied:
Alright, I'll fix it.

helish88 replied:
it's ok right now, why are you trying to commit with errors kek

helish88 commented on main.py:
```diff
@@ -57,63 +66,68 @@ def check(m):
     logembed.set_thumbnail(url=ctx.author.avatar_url)
 
-    logchannel = ctx.guild.get_channel(609431364445405194)  # Put your channel, you would like to send giveaway logs to.
+    guild = client.get_guild(796081913688883240)  # Put your guild ID here!
```
client?

blobonmyknob replied:
Sorry, I usually use client, it's more updated from the word bot.

helish88 commented on main.py:
```diff
@@ -124,45 +138,129 @@ def check(m):
 
+    win = await msg.edit(embed = discord.Embed(title = "WINNER" , description = f"Congratulations {winnerstosend}, you have won **{msg4.content}**!" , color = discord.Color.blue()))
+    
+    print(Fore.GREEN + winnerstosend , Fore.WHITE + "has won: " , Fore.GREEN + msg4.content)
```
I don't think we need a prints here

blobonmyknob replied:
It's just to make the terminal look better, all github open source over does it.

helish88 replied:
If you host bot on VPS you will not see prints

blobonmyknob replied:
it's alright then vps or not, if they have it it will show I guess.

blobonmyknob committed:
Update main.py

Fixed errors

blobonmyknob reviewed (commented):
it should be fixed now

helish88 commented on main.py:
```diff
@@ -57,63 +66,68 @@ def check(m):
-                           title=f"🎉GIVEAWAY🎉\n`{msg4.content}`", timestamp=futuredate,
-                           description=f'React with 🎉 to enter!\nHosted by: {ctx.author.mention}')
-
+    embed1 = discord.Embed(color = discord.Color(random.randint(0x000000, 0xFFFFFF)), title=f"🎉GIVEAWAY🎉\n`{msg4.content}`",timestamp = futuredate, description=f'React with 🎉 to enter!\nHosted by: {ctx.author.mention}')
```
same code

blobonmyknob replied:
This is just what it looks like on mine, I did a pull request for additional print statements, has_permissions decorators and error decorators.

helish88 replied:
> 
> 
> I added all those error messages, for missing permissions.
> ![image](https://user-images.githubusercontent.com/76411422/106044839-e43ceb00-60a5-11eb-8446-e8b0335e41cd.png)
> I also added error messages for commands that are upcoming.
> ![image](https://user-images.githubusercontent.com/76411422/106044903-f74fbb00-60a5-11eb-9e1c-255d4333b35b.png)
> If your questioning if I'm pulling my own weight, I think I am, I made an effort to create the terminal better then what it was just for aesthetic purposes, and I also took the time to create error messages for people who don't have the right permissions instead of the bot just returning nothing.
> 
> Think of it this way without my pull requests anybody can use the bot and start a giveaway, now they need permissions, and the bot will tell them if they don't. @helish88

I think manage_roles is enough

blobonmyknob replied:
I suggest theese commands have manage guild permissions needed to use it:
``!reroll``
``!pickwinner`` (Still being developed by me.)
I will change it.

blobonmyknob commented:
It should be all fixed now @helish88 
The print statements just won't show if they are hosting the bot, but it will show if they are, there is no need to remove it I think.

blobonmyknob commented on main.py:
```diff
@@ -124,45 +138,129 @@ def check(m):
+    
+    
+# Command Error Events.
+@giveaway.error
```
Look at all this

blobonmyknob commented on main.py:
```diff
@@ -124,45 +138,129 @@ def check(m):
+async def reroll_error(ctx, error):
+    if isinstance(error, MissingPermissions):
+        await ctx.send(f"ERROR: Hello {ctx.author}, it seems you have encountered an error.\nYou are missing the following permissions: ``Manage Server``\nYou cannot reroll the giveaway without theese permissions.")
+        print(Fore.WHITE + ctx.author , 'encountered an error:\n' , Fore.RED + 'Missing Permissions\n' , Fore.WHITE + "Command:" , Fore.GREEN + 'Reroll')
```
down to here

blobonmyknob commented:
I added all those error messages, for missing permissions.
![image](https://user-images.githubusercontent.com/76411422/106044839-e43ceb00-60a5-11eb-8446-e8b0335e41cd.png)
I also added error messages for commands that are upcoming.
![image](https://user-images.githubusercontent.com/76411422/106044903-f74fbb00-60a5-11eb-9e1c-255d4333b35b.png)
If your questioning if I'm pulling my own weight, I think I am, I made an effort to create the terminal better then what it was just for aesthetic purposes, and I also took the time to create error messages for people who don't have the right permissions instead of the bot just returning nothing.

Think of it this way without my pull requests anybody can use the bot and start a giveaway, now they need permissions, and the bot will tell them if they don't. @helish88

blobonmyknob commented:
Also can we please switch from **bot** to **client** when defining our bot, it just looks bad.
Most people use client, from when rewrite came out, they mean the same thing but in a way client means both.

It's just for aesthetic purposes, don't call me weird but I tend to get a little scared when I see the word **bot** defined in decorators, it's just reminds me about discord.py before rewrite, which was absolute hell.

blobonmyknob committed:
Update main.py

helish88 commented:
I'm confused, make a new commit
